image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -0,0 +1,138 @@
+       Therefore, we use this explicitly set configuration. */}}
+  {{- if not (kindIs "invalid" $scannerV4Cfg.indexer.disable) -}}
+    {{- $_ = set $components "indexer" (not $scannerV4Cfg.indexer.disable) -}}
+  {{- end -}}
+  {{- if not (kindIs "invalid" $scannerV4Cfg.matcher.disable) -}}
+    {{- $_ = set $components "matcher" (not $scannerV4Cfg.matcher.disable) -}}
+  {{- end -}}
+
+  {{/* This is potentially an unsupported configuration. Hence we generate a warning. */}}
+  {{- $msg := printf "indexer is %s and matcher is %s." (ternary "enabled" "disabled" (get $components "indexer")) (ternary "enabled" "disabled" (get $components "matcher")) -}}
+  {{- if eq $.Chart.Name "stackrox-central-services" -}}
+    {{- if or (ne (get $components "indexer") (get $componentsCentralChart "indexer")) (ne (get $components "matcher") (get $componentsCentralChart "matcher")) -}}
+      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
+    {{- end -}}
+  {{- else -}}
+    {{- if or (ne (get $components "indexer") (get $componentsSecuredClusterChart "indexer")) (ne (get $components "matcher") (get $componentsSecuredClusterChart "matcher")) -}}
+      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
+    {{- end -}}
+  {{- end -}}
+
+{{- end -}}
+
+{{/* Configure images, certificates and passwords as required. */}}
+{{/* Note that for secured-cluster-services we don't configure certificates here, instead they will be distributed at runtime by Sensor and Central. */}}
+{{ if get $components "indexer" }}
+  {{ include "srox.configureImage" (list $ $scannerV4Cfg.indexer.image) }}
+  {{- if eq $.Chart.Name "stackrox-central-services" -}}
+    {{/* Only generate certificate when installing central-services. */}}
+    {{ $cryptoSpec := dict "CN" "SCANNER_SERVICE: Scanner V4 Indexer" "dnsBase" "scanner-v4" }}
+    {{ include "srox.configureCrypto" (list $ "scannerV4.indexer.serviceTLS" $cryptoSpec) }}
+  {{- end -}}
+{{ end }}
+
+{{ if get $components "matcher" }}
+  {{ include "srox.configureImage" (list $ $scannerV4Cfg.matcher.image) }}
+  {{- if eq $.Chart.Name "stackrox-central-services" -}}
+    {{/* Only generate certificate when installing central-services. */}}
+    {{ $cryptoSpec := dict "CN" "SCANNER_SERVICE: Scanner V4 Matcher" "dnsBase" "scanner-v4" }}
+    {{ include "srox.configureCrypto" (list $ "scannerV4.matcher.serviceTLS" $cryptoSpec) }}
+  {{- end -}}
+{{ end }}
+
+{{ if or (get $components "indexer") (get $components "matcher") }}
+  {{ include "srox.configureImage" (list $ $scannerV4Cfg.db.image) }}
+  {{ include "srox.configurePassword" (list $ "scannerV4.db.password") }}
+  {{- if eq $.Chart.Name "stackrox-central-services" -}}
+    {{/* Only generate certificate when installing central-services. */}}
+    {{ $cryptoSpec := dict "CN" "SCANNER_SERVICE: Scanner V4 DB" "dnsBase" "scanner-v4-db" }}
```

Comment:
Currently, we have `SCANNER_DB_SERVICE` for Scanner DB. Let's use that here, too